manim/mobject/svg/svg_mobject.py

```diff
@@ -126,116 +123,142 @@ def generate_points(self):
         else:
             pass  # TODO
-            # warnings.warn("Unknown element type: " + element.tagName)
+
         result = [m for m in result if m is not None]
         self.handle_transforms(element, VGroup(*result))
         if len(result) > 1 and not self.unpack_groups:
             result = [VGroup(*result)]
 
-        return result
-
-    def g_to_mobjects(self, g_element):
-        """Converts the ``g`` SVG element into VMobjects.
+        if within_defs and element.hasAttribute("id"):
+            self.def_id_to_mobject[element.getAttribute("id")] = result
+        if is_defs:
+            # defs shouldn't be part of the result tree, only the id dictionary.
+            return []
 
-        Parameters
-        ----------
-        g_element : :class:`str`
-            A ``g`` element is a group of other SVG elements. As such a ``g`` element is equivalent to a VGroup.
-
-        Returns
-        -------
-        List[VMobject]
-            A list of VMobject represented by the group.
-        """
-        mob = VGroup(*self.get_mobjects_from(g_element))
-        self.handle_transforms(g_element, mob)
-        return mob.submobjects
+        return result
 
-    def path_string_to_mobject(self, path_string):
+    def path_string_to_mobject(self, path_string: str, style: dict):
         """Converts a SVG path element's ``d`` attribute to a mobject.
 
         Parameters
         ----------
-        path_string : str
+        path_string : :class:`str`
             A path with potentially multiple path commands to create a shape.
 
+        style : :class:`dict`
+            Style specification, using the SVG names for properties.
+
         Returns
         -------
         VMobjectFromSVGPathstring
             A VMobject from the given path string, or d attribute.
         """
-        return VMobjectFromSVGPathstring(path_string)
+        return SVGPathMobject(path_string, **parse_style(style))
 
-    def use_to_mobjects(self, use_element):
-        """Converts a SVG <use> element to VMobject.
+    def use_to_mobjects(self, use_element: MinidomElement) -> List[VMobject]:
+        """Converts a SVG <use> element to a collection of VMobjects.
 
         Parameters
         ----------
-        use_element : str
+        use_element : :class:`MinidomElement`
             An SVG <use> element which represents nodes that should be
             duplicated elsewhere.
 
         Returns
         -------
-        VMobject
-            A VMobject
+        List[VMobject]
+            A collection of VMobjects that are copies of the defined objects
         """
+
         # Remove initial "#" character
         ref = use_element.getAttribute("xlink:href")[1:]
-        if ref not in self.ref_to_element:
-            warnings.warn("%s not recognized" % ref)
-            return VGroup()
-        return self.get_mobjects_from(self.ref_to_element[ref])
+
+        try:
+            return [i.copy() for i in self.def_id_to_mobject[ref]]
```

Comment:
I would prefer using the `copy` module here rather than `i.copy()`, also is `i` a Mobject here?

Reply:
That is correct, `i` is a mobject. What's the correct way to use the copy module? I'm unfamiliar with it.

Reply:
Actually copying something should be done using [copy](https://docs.python.org/3/library/copy.html), usually, python object don't `.copy()` functions, of course, manim provides, but using that module seems to be more appropriate to me atleast

Reply:
but it's ok, in the future I would like it to be depreciated.

Reply:
ah, so something like `return [copy.deepcopy(i) for i in self.def_id_to_mobject[ref]]` ? If so, under the hood it looks like that's the case:
https://github.com/ManimCommunity/manim/blob/4237e00edc4f43739ac02d508a375a01a006d69d/manim/mobject/mobject.py#L463-L464

Reply:
it's a niche method for users, actually, but it's ok to use though.